site/_includes/render-menu.html

```diff
@@ -0,0 +1,93 @@
+{%- comment -%}
+Variables pass to this file:
+  menuName: The name of the menu to be rendered. Default: "primary"
+  tag:      HTML Tag to surround Menu name.  Default: false (No Tags)
+  displayName: Boolean.  Whether to display menu name or not. Default: false
+
+NOTE: The variable mName was originally labeled menuName.  For reasons that
+are still unclear, menuName would not change when assigned a new value.
+This resulted in it always being the same menu and causing an infinite
+recursive loop to occur.  The followiing error message occurred when
+generating static files:
+Liquid Exception: Liquid error (/site/_includes/render-menu.html line 97): Nesting too deep included in menus.html
+If this error occurs in the future (file name may change), then there is a problem with the mr-menus.yml file.
+{%- endcomment -%}
+
+{%- comment -%}
+If the include.menuName is not present or is an empty string, it is assigned to
+be the Primary Menu.
+
+If the include.menuName is "primary", "secondary", "footer" then substitute 
+the real menu name that has been assigned to the placement menu variable.
+{%- endcomment -%}
+{%- if include.menuName == nil or include.menuName == blank or include.menuName == "primary" -%}
+    {%- assign mName = site.data.mr-menus.primary -%}
+{%- elsif include.menuName == "secondary" -%}
+    {%- assign mName = site.data.mr-menus.secondary -%}
+{%- elsif include.menuName == "footer" -%}
+    {%- assign mName = site.data.mr-menus.footer -%}
+{%- else -%}
+    {%- assign mName = include.menuName -%}
```

Comment:
The `site.data.mr_menus` really needs an underscore and not a hyphen.

See the test cases in [features/data.feature](https://github.com/jekyll/jekyll/blob/master/features/data.feature#L1) from Jekyll's source. It works here, but may get messy. 

$`\frac{convention}{configuration}`$

The code for how Jekyll merges the contents of `_data` is in [lib/jekyll/document.rb](https://github.com/jekyll/jekyll/blob/master/lib/jekyll/document.rb#L13-L18) and [lib/jekyll/utils.rb](https://github.com/jekyll/jekyll/blob/master/lib/jekyll/utils.rb#L13-L47)

The regexps it uses for conversion/interpretation of slugs, keys, etc is all in there.

Reply:
i've worked with ruby a lot. i just don't want to debug something like that. it's easier this way.

Reply:
In this case, the use of underscore is the wrong choice.  All the CSS/SCSS from Basically Basic uses hyphens instead of underscore.  Therefore, to switch to underscore is breaking with the convention already set because we are using and starting with  the Basically Basic theme's naming convention.  Besides, there has been no actual decision to use snake naming convention.

Reply:
In any framework we use, especially when they involve metaprogramming, we need to be clear on how identifiers should be cased. 

I can find/replace things very quickly. I just don't want to argue about this. I can be wrong. The case & context should underscore the purpose of a variable or identifier. 

If things get muddied, then doing large find/replace is more error prone.